core/src/software/aws/toolkits/core/ToolkitClientManager.kt

```diff
@@ -132,7 +136,14 @@ abstract class ToolkitClientManager {
                         builder.endpointOverride(URI.create(it))
                     }
                 }
+                .apply(clientCustomizer)
                 .build() as T
         }
     }
 }
+
+fun <T : SdkClient> ToolkitClientManager.getClient(
```

Comment:
We are probably getting to the point where we need to decide if we want to go all in on ConnectionSettings and kill off the older APIs.

Reply:
I believe we should - anywhere we pass the `region` / `credential` pair we should just replace it with `ConnectionSettings' 

This was the first step in that direction - the above will be a much bigger refactor (and can probably be done piecemeal)

Reply:
Agree. It is larger than this PR